src/ring_theory/fractional_ideal.lean

```diff
@@ -103,69 +101,76 @@ variables [algebra R P] [loc : is_localization S P]
 
 instance : has_coe (fractional_ideal S P) (submodule R P) := ⟨λ I, I.val⟩
 
-@[simp] lemma val_eq_coe (I : fractional_ideal S P) : I.val = I := rfl
+protected lemma is_fractional (I : fractional_ideal S P) :
+  is_fractional S (I : submodule R P) :=
+I.prop
 
-@[simp, norm_cast] lemma coe_mk (I : submodule R P) (hI : is_fractional S I) :
-  (subtype.mk I hI : submodule R P) = I := rfl
+section set_like
 
-instance : has_mem P (fractional_ideal S P) := ⟨λ x I, x ∈ (I : submodule R P)⟩
+instance : set_like (fractional_ideal S P) P :=
+{ coe := λ I, ↑(I : submodule R P),
+  coe_injective' := set_like.coe_injective.comp subtype.coe_injective }
 
-/-- Fractional ideals are equal if their submodules are equal.
+@[ext] lemma ext {I J : fractional_ideal S P} : (∀ x, x ∈ I ↔ x ∈ J) → I = J := set_like.ext
 
-  Combined with `submodule.ext` this gives that fractional ideals are equal if
-  they have the same elements.
--/
-lemma coe_injective {I J : fractional_ideal S P} : (I : submodule R P) = J → I = J :=
-subtype.ext_iff_val.mpr
+/-- Copy of a `fractional_ideal` with a new underlying set equal to the old one.
+Useful to fix definitional equalities. -/
+protected def copy (p : fractional_ideal S P) (s : set P) (hs : s = ↑p) : fractional_ideal S P :=
+⟨submodule.copy p s hs, by { convert p.is_fractional, ext, simp only [hs], refl }⟩
+
+end set_like
+
+@[simp] lemma val_eq_coe (I : fractional_ideal S P) : I.val = I := rfl
 
-lemma ext_iff {I J : fractional_ideal S P} : (∀ x, (x ∈ I ↔ x ∈ J)) ↔ I = J :=
-⟨λ h, coe_injective (submodule.ext h), λ h x, h ▸ iff.rfl⟩
+@[simp, norm_cast] lemma coe_mk (I : submodule R P) (hI : is_fractional S I) :
+  (subtype.mk I hI : submodule R P) = I := rfl
 
-@[ext] lemma ext {I J : fractional_ideal S P} : (∀ x, (x ∈ I ↔ x ∈ J)) → I = J :=
-ext_iff.mp
+@[simp, norm_cast] lemma mem_coe {I : fractional_ideal S P} {x} :
+  x ∈ (I : submodule R P) ↔ x ∈ I := iff.rfl
 
-lemma fractional_of_subset_one (I : submodule R P)
-  (h : I ≤ (submodule.span R {1})) :
+lemma coe_to_submodule_injective :
```

Comment:
Possibly worth a comment like "not to be confused with `whatever_the_namespace_is.coe_submodule`"

Reply:
I added clarifying comments to the definitions of the `has_coe` instances.